src/components/ContactsAdd.js

```diff
@@ -1,39 +1,144 @@
-import { useState } from "react"
-import { useNavigate } from "react-router-dom";
+import { useState } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+
+const initialData = {
+  firstName: "",
+  lastName: "",
+  street: "",
+  city: "",
+  email: "",
+  truthSocial: "",
+  linkedIn: "",
+};
 
 function ContactsAdd(props) {
+  const contacts = props.contacts;
+  const setContacts = props.setContacts;
+  const [inputData, setInputData] = useState(initialData);
+  const navigate = useNavigate();
+
+  const handleChange = (ev) => {
+    const { name, value, type } = ev.target;
+
+    if (name === "firstName" && type === "text") {
+      setInputData({ ...inputData, firstName: value });
+    }
+    if (name === "lastName" && type === "text") {
+      setInputData({ ...inputData, lastName: value });
+    }
+    if (name === "street" && type === "text") {
+      setInputData({ ...inputData, street: value });
+    }
+    if (name === "city" && type === "text") {
+      setInputData({ ...inputData, city: value });
+    }
+    if (name === "email" && type === "email") {
+      setInputData({ ...inputData, email: value });
+    }
+    if (name === "linkedIn" && type === "text") {
+      setInputData({ ...inputData, linkedIn: value });
+    }
+    if (name === "truthSocial" && type === "text") {
+      setInputData({ ...inputData, truthSocial: value });
+    }
+  };
+
+  const handleSubmit = (ev) => {
+    ev.preventDefault();
 
-  // setContacts and contacts must be passed as props
-  // to this component so new contacts can be added to the
-  // state
-  const { setContacts, contacts } = props
+    fetch(`http://localhost:4000/contacts`, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(inputData),
+    })
+      .then((res) => res.json(inputData))
+      .then((data) => {
+        setContacts([...contacts, data]);
+        setInputData(initialData);
+        navigate("/contacts");
```

Comment:
This is great, as soon as the data is fetched the and added to the contact user state, it takes you to the contact page on submit